### ISSUE TITLE
CatBoost unexpected output on simple synthetic dataset

### ISSUE THREAD

**Issue by gitmario** [catboost/catboost#1141]
Hello! I've just set up a simple example in which CatBoost behaves in a strange way. The training set consists in 300 data points, and the model should learn this simple function:
"A" -> 10
"B" -> 20
"C" -> 30

The model gives the correct output for `"A"` (according to training examples) and `"D"` (handling an unseen level, I get approximately the average label on the training set)

However, for `"B"` and `"C"` the model fails to learn the simple function.

```
from catboost import CatBoostRegressor

# Initialize data
train_data = [["A"],
              ["B"],
              ["C"]] * 100

train_labels = [10, 20, 30] * 100

# Initialize CatBoostRegressor
model = CatBoostRegressor(cat_features=[0])

# Fit model
model.fit(train_data, train_labels, verbose=False)

print("{:.2f}".format(model.predict(['A'])))
print("{:.2f}".format(model.predict(['B'])))
print("{:.2f}".format(model.predict(['C'])))
print("{:.2f}".format(model.predict(['D'])))
```

Output:
```
10.00
25.00
25.00
19.99
```

Expected output:
```
10.00
20.00
30.00
20.00
```

`(catboost==0.20.1)`

**Comment by gitmario** [comment]
Here's a similar example, this time with numeric features: handling a missing value, I would expect the average label of the training set as output. Now the model should learn this function:
1 -> 10
2 -> 20
3 -> 30

I would expect
{missing input} -> 20
instead of
{missing input} -> 10

```
import numpy as np
from catboost import CatBoostRegressor

# Initialize data
train_data = [1,
              2,
              3] * 100

train_labels = [10, 20, 30] * 100

# Initialize CatBoostRegressor
model = CatBoostRegressor()

# Fit model
model.fit(train_data, train_labels, verbose=False)

print("{:.2f}".format(model.predict([1])))
print("{:.2f}".format(model.predict([2])))
print("{:.2f}".format(model.predict([3])))
print("{:.2f}".format(model.predict([np.nan])))
```

Output:
```
10.00
20.00
30.00
10.00
```

Expected Output:
```
10.00
20.00
30.00
20.00
```

**Comment by kizill** [comment]
In case of float float features, we support two ways of processing NaN values. Treat them as -inf, or as +inf, depending on `nan_mode` parameter. If there were no NaN values in train dataset, they would work as -inf value in prediction.
